platforms/hermes-javascript/README.md

```diff
@@ -0,0 +1,654 @@
+
+## Setup
+
+```sh
+npm install hermes-javascript
+```
+
+`Hermes-javascript` uses a dynamic library generated by the hermes rust code under the hood.
+The installation process will automagically download the file if your os and architecture is supported.
+
+#### ⚠️ Unsupported platforms / architectures
+
+If the setup could not infer the library file version, it will attempt to build it from the sources.
+Please note that [`rust`](https://www.rust-lang.org/tools/install) and [`git`](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) are required in order to build the library!
+
+If you want to force this behaviour, you can also define the `HERMES_BUILD_FROM_SOURCES` environment variable before running `npm install`.
+
+```sh
+env HERMES_BUILD_FROM_SOURCES=true npm install hermes-javascript
+```
+
+## Usage
+
+### Minimal use case
+
+```js
+const { withHermes } = require('hermes-javascript')
+
+/*
+    A small js context manager that sets up an infinite loop to prevent
+    the process from exiting, and exposes an instance of the Hermes class.
+*/
+withHermes(hermes => {
+    // Instantiate a dialog object
+    const dialog = hermes.dialog()
+
+    // Subscribes to intent 'myIntent'
+    dialog.flow('myIntent', (msg, flow) => {
+        // Log intent message
+        console.log(JSON.stringify(msg))
+        // End the session
+        flow.end()
+        // Use text to speech
+        return `Received message for intent ${myIntent}`
+    })
+})
+```
+
+### Expanded use case
+
+```js
+const { withHermes } = require('hermes-javascript')
+
+/*
+    The 'withHermes' function exposes a done() function
+    that can be called to clean up the context loop and exit.
+*/
+withHermes((hermes, done) => {
+    // NB: Dialog is only one of the available API Subsets.
+    const dialog = hermes.dialog()
+
+    /*
+        Every API Subset can publish and receive data based on a list of events.
+
+        For the purpose of this example, we will only use the Dialog subset, and the
+        events related to a dialog session.
+
+        Note that more events are available for each subset.
+    */
+
+    // You can subscribe to an event triggered when the intent 'intentName' is detected like this:
+    dialog.on('intent/intentName', message => {
+
+        // The 'message' argument contain all the data you need to perform an action based on what the user said.
+
+        // For instance, you can grab a slot and its value like this.
+        const mySlot = msg.slots.find(slot => slot.slot_name === 'slotName')
+        const slotValue = mySlot.value.value
+        // And here is how to grab the intent name.
+        console.log('Received intent', message.intent.intent_name)
+
+        // Then, you can either :
+        // 1 - Continue the dialog session if you expect another intent to be detected.
+        dialog.publish('continue_session', {
```

Comment:
maybe wrap the two `dialogue.publish` in a if/else, otherwise strange things may happen if an user just blindly copy paste the code without reading the comments

Reply:
✅